### ISSUE TITLE
Subsequent manoeuvres don't honour distance formatting configuration

### ISSUE THREAD

**Issue by sandyscoffable** [mapbox/mapbox-navigation-android#4476]
**Android API:** 30
**Mapbox Navigation SDK version:** 2.0.0-beta.12

### Steps to trigger behavior

 1. Configure `MapboxManeuverApi` as follows
 ```kotlin
maneuverApi = MapboxManeuverApi(
   MapboxDistanceFormatter(
     DistanceFormatterOptions.Builder(requireContext()).locale(Locale.UK).unitType(UnitType.IMPERIAL).build()
   )
)
 ```
 2. Set up a route with some turns

### Expected behavior

All distances should be represented as miles / feet

### Actual behavior

Only the first maneuver has miles / feet. Subsequent maneuvers don't. i.e
<img src="https://user-images.githubusercontent.com/20515901/120796092-4a44ed80-c532-11eb-8ff7-129e66c2a14a.png" width="320" />

**Comment by LukasPaczos** [comment]
cc @abhishek1508

**Comment by abhishek1508** [comment]
This is interesting. I haven't seen this ever. Thanks for creating the issue @sandyscoffable 
Will look into it.

**Comment by sandyscoffable** [comment]
I wondered if there was somewhere else I needed to be setting this, so I tried setting the same `DistanceFormatterOptions` object in the higher level `NavigationOptions` object (used when you initialise MapboxNavigation) in the hope that this would fix the issue, but that had no effect.

**Comment by dzinad** [comment]
I was able to reproduce the issue with 2.0.0-beta.12, but it was fixed in the stable 2.0.0 release. The API changed between beta.12 and the stable release, eliminating the possibility of dissynchronization.
The first screenshot represents v2.0.0-beta.12, the second one - v2.0.0.
![photo_2022-08-29 19 11 07](https://user-images.githubusercontent.com/32109537/187246042-6cdd5ebb-4311-4adb-978c-e48268b866b2.jpeg)
![photo_2022-08-29 19 11 12](https://user-images.githubusercontent.com/32109537/187246065-0b8e686a-1a72-481d-a5a5-8a4e8debbea1.jpeg)
